contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -1033,16 +1033,49 @@ update_GrantStmt(Node *n, const char *object, const char *obj_schema, const char
 	else if (obj_schema && stmt->objects)
 	{
 		RangeVar   *tmp = (RangeVar *) llast(stmt->objects);
-
 		tmp->schemaname = pstrdup(obj_schema);
 	}
 
 	if (grantee && stmt->grantees)
 	{
 		RoleSpec   *tmp = (RoleSpec *) llast(stmt->grantees);
-
+		if (strcmp(grantee, "public") == 0)
+		{
+			tmp->roletype = ROLESPEC_PUBLIC;
+		}
 		tmp->rolename = pstrdup(grantee);
 	}
+
+	if (priv && stmt->privileges)
+	{
+		AccessPriv *tmp = (AccessPriv *) llast(stmt->privileges);
+		tmp->priv_name = pstrdup(priv);
+	}
+}
+
+void
+update_AlterDefaultPrivilegesStmt(Node *n, const char *object, const char *grantee, const char *priv)
+{
+	AlterDefaultPrivilegesStmt *stmt = (AlterDefaultPrivilegesStmt *) n;
+	ListCell *lc;
+
+	if (!IsA(stmt, AlterDefaultPrivilegesStmt))
+		ereport(ERROR, (errcode(ERRCODE_SYNTAX_ERROR), errmsg("query is not a AlterDefaultPrivilegesStmt")));
+
+	if (grantee && priv && stmt->action)
+	{
+		update_GrantStmt((Node *)(stmt->action), NULL, NULL, grantee, priv);
+	}
+
+	foreach(lc, stmt->options)
+	{
+		if (object)
+		{
+			DefElem *tmp = (DefElem *) lfirst(lc);
+			tmp->defname = pstrdup("schemas");
```

Comment:
should we free tmp->defm=name before allocating new memory?

Reply:
I verified and we don't need to pfree() this pointer explicitly since it's already NULL at this point.